java/se/sics/mspsim/core/RTC.java

```diff
@@ -562,7 +562,7 @@ public void write(int address, int value, boolean word, long cycles) {
                                 parseCalReg(Calendar.SECOND, lo);
                                 parseCalReg(Calendar.MINUTE, hi);
                         } else {
-                                rtcCount &= 0xffff0000;
+                                rtcCount &= 0xffff_ffff_ffff_0000L;
```

Comment:
I am not that familiar with the RTC code but I think it looks like the RTC count is an unsigned 32-bit value stored as long, probably to make sure it never becomes negative. If so, I think the intention might have been `rtcCount &= 0xffff0000L`. See line 575 for a comparison.

Reply:
Fixed.